tasks/libs/junit_upload_core.py

```diff
@@ -217,6 +220,14 @@ def upload_junitxmls(output_dir, owners, flavor, xmlfile_name, process_env, addi
             raise subprocess.CalledProcessError(exit_code, DATADOG_CI_COMMAND)
 
 
+def is_e2e_internal_failure(xml_path):
+    """
+    Check if the given JUnit XML file contains E2E INTERAL ERROR string.
+    """
+    filecontent = open(xml_path).read()
+    return "E2E INTERNAL ERROR" in filecontent
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
`"E2E INTERNAL ERROR"` could be a global constant